src/python/ddapp/tasks/robottasks.py

```diff
@@ -706,6 +708,15 @@ def run(self):
         side = self.properties.getPropertyEnumValue('Side').lower()
         self.getHandDriver(side).sendCustom(self.properties.getProperty('Amount'), 100, 100, self.properties.getProperty('Mode'))
 
+        if self.properties.getProperty('Check status'):
+            responseChannel = 'GRASPING_STATE'
+            responseMessageClass = lcmdrc.boolean_t
+            grasping_state = lcmUtils.MessageResponseHelper(responseChannel, responseMessageClass).waitForResponse(timeout=7000)
+            if grasping_state is not None and grasping_state.data == 1:
+                self.statusMessage = "Grasping successful"
+            else:
+                self.fail("No object in hand")
+
```

Comment:
You could keep the 'Check status' property here.  The value of 'Check status' determines whether or not WaitForGraspingState is invoked here.  So the behavior would be exactly the same.

Or, if you wanted, you could remove the 'Check status' property, and instead the task queue could have a CloseHand task followed by a WaitForGraspingState task.  In some ways, this strategy might be more clear.  Because the CloseHand task succeeds, but  the failure might be during WaitForGraspingState.  Of course, then 'resuming' the task queue would resume at the WaitForGraspingState, when you might really want to resume at CloseHand (to retry).  So that argues for keeping CloseHand and Wait a single task, as you currently have it.